### PR TITLE
Add Clojure

### PR DESCRIPTION
**Pull request by buob**

### CODE REVIEW

File: core/src/main/web/plugin/init/addevalplugins.js [hunk 1]
```diff
@@ -25,7 +25,8 @@
       "Java": { url : "./plugins/eval/javash/javash.js", bgColor: "#EB0000", fgColor: "#FFFFFF", borderColor: "", shortName: "Jv" },
       "R": { url : "./plugins/eval/r/r.js", bgColor: "#8495BB", fgColor: "#FFFFFF", borderColor: "", shortName: "R" },
       "Scala": { url : "./plugins/eval/scala/scala.js", bgColor: "#B41703", fgColor: "#FFFFFF", borderColor: "", shortName: "Sc" },
+      "Clojure": { url : "./plugins/eval/clojure/clojure.js", bgColor: "#DB5855", fgColor: "#FFFFFF", borderColor: "", shortName: "Cl" },
```
Comment (scottdraves) [inline core/src/main/web/plugin/init/addevalplugins.js]: use the dark blue color from the clojure logo 5881d8

File: plugin/clojure/src/dist/clojure.js [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+  var PLUGIN_NAME = "Clojure";
+  var COMMAND = "clojure/clojurePlugin";
+  var serviceBase = null;
+  var cometdUtil = bkHelper.getUpdateService();
+  var cancelFunction = null;
+
+  var Clojure = {
+      pluginName: PLUGIN_NAME,
+      cmMode: "text/x-clojure",
+      background: "#DB5855",
```
Comment (scottdraves) [inline plugin/clojure/src/dist/clojure.js]: ditto

File: plugin/clojure/src/dist/clojurePlugin [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import os
+import sys
+
+port = sys.argv.pop()
+self_path = sys.argv.pop(0)
+
+os.chdir(os.path.dirname(self_path))
+
+os.environ["CLOJURE_OPTS"] = " ".join(sys.argv)
```
Comment (scottdraves) [inline plugin/clojure/src/dist/clojurePlugin]: i don't think CLOJURE_OPTS is a thing is it?

**Comment by scottdraves** [comment]
Looks great Jake, just a few tiny changes.

File: core/src/main/web/plugin/init/addevalplugins.js (hunk 1 above)
Comment (scottdraves) [inline core/src/main/web/plugin/init/addevalplugins.js]: shortname to "Cj"

**Comment by ildipo** [comment]
It looks good, but it is still missing auto translation and interface with the notebook.
To add it you should use the NamespaceClient object. At evaluator initialization you should set something in the global namespace that points to the instance of NamespaceClient, as it is done for groovy and scala. Just before executing the code you update the bindings of it with the job under evaluation.

**Comment by ildipo** [comment]
Similar thing is for stdout and stderr - does clojure have a way to output it? did you try to see if it is redirected properly? I see the code it is there.

**Comment by ildipo** [comment]
What about autocompletion? Is there a way to list the environment? Are there classes in the REPL implementation you can leverage?

**Comment by ildipo** [comment]
Imports and class path? Class path probably it is easy - just mimic what it is done by groovy and scala with the DynamicClassLoader and the two options in the plugin interface. Imports might be transformed in the clojure equivalent. It would be nice to have a way of producing plots from clojure.

**Comment by ildipo** [comment]
Finally we need an example notebook. :-)

**Comment by scottdraves** [comment]
i am seeing failure to start randomly, with this output on console:

Failed to load resource: the server responded with a status of 404 (Not Found)
http://127.0.0.1:8801/9d31ada/clojure.788067533904/rest/clojuresh/getShell Failed to load resource: the server responded with a status of 404 (Not Found)
:8801/9d31ada/beaker/plugins/eval/clojure/clojure.js:43 failed to create shell Arguments[4]
http://127.0.0.1:8801/9d31ada/clojure.788067533904/cometd/handshake?jsonp=j…3A60000%2C%22interval%22%3A0%7D%2C%22id%22%3A%222%22%7D%5D&_=1426855362548 Failed to load resource: the server responded with a status of 404 (Not Found)
http://127.0.0.1:8801/9d31ada/clojure.788067533904/cometd/handshake?jsonp=j…3A60000%2C%22interval%22%3A0%7D%2C%22id%22%3A%223%22%7D%5D&_=1426855362549 Failed to load resource: the server responded with a status of 404 (Not Found)

**Comment by scottdraves** [comment]
reproduced the failure on another computer
screen shot 2015-03-20 at 9 50 27 am

**Comment by ildipo** [comment]
I can run it. Did not see the startup issue (yet).
Also stdout redirect works perfectly already.

**Comment by scottdraves** [comment]
:dolphin:

**Pull request merged**